### PR TITLE
Add python script and quick start guide for MetricsReloaded

### PR DESCRIPTION
**Pull request by valosekj**
## Description

This PR adds:
- [MetricsReloaded_quick_start_guide.md](https://github.com/ivadomed/utilities/blob/3d0bce83488453eb5f454e506bb4555591735151/quick_start_guides/MetricsReloaded_quick_start_guide.md) providing instructions on how to install [MetricsReloaded](https://github.com/ivadomed/MetricsReloaded) and use it
- [compute_metrics_reloaded.py](https://github.com/ivadomed/utilities/blob/3d0bce83488453eb5f454e506bb4555591735151/compute_metrics/compute_metrics_reloaded.py) to compute segmentation metrics using [MetricsReloaded](https://github.com/ivadomed/MetricsReloaded/tree/main) and save them into a CSV file

## Useful links:

- [MetricsReloaded GitHub page](https://github.com/Project-MONAI/MetricsReloaded)
- [Our ivadomed fork](https://github.com/ivadomed/MetricsReloaded/tree/main)
- [MetricsReloaded documentation](https://metricsreloaded.readthedocs.io/en/latest/)
- [MetricsReloaded publication](https://www.nature.com/articles/s41592-023-02151-z)
- [MetricsReloaded preprint](https://arxiv.org/pdf/2206.01653v5.pdf) - preprint contains more figures than the publication

### CODE REVIEW

**Comment by valosekj** [comment]
When digging into the [MetricsReloaded](https://github.com/Project-MONAI/MetricsReloaded/tree/main) code, @naga-karthik and I did the following:

- we added missing `vol_diff` key to `BinaryPairwiseMeasures.measures_dict` and opened [PR](https://github.com/Project-MONAI/MetricsReloaded/pull/44)
- I forked the repo and added a new `Relative Volume Error` metric in [this PR](https://github.com/valosekj/MetricsReloaded/pull/1)

**Comment by valosekj** [comment]
Edge cases (using [this PR](https://github.com/valosekj/MetricsReloaded/pull/1)):

1. Empty reference **and** empty prediction --> model learned correctly

```json
{
    "reference": "ref.nii.gz",
    "prediction": "pred.nii.gz",
    "0": {
        "dsc": 1,
        "fbeta": 1,
        "nsd": NaN,
        "vol_diff": NaN,
        "rel_vol_diff": 0,
        "EmptyRef": true,
        "EmptyPred": true
    }
}
```

2. Empty reference **and not** empty prediction --> model did not learn correctly (assuming that the GT is correct) --> model is over-segmenting

```json
{
    "reference": "ref.nii.gz",
    "prediction": "pred.nii.gz",
    "1.0": {
        "dsc": 0.0,
        "fbeta": 0,
        "nsd": 0.0,
        "vol_diff": Infinity,
        "rel_vol_diff": 100,
        "EmptyRef": true,
        "EmptyPred": false
    }
}
```

3. **Non**-empty reference **and** empty prediction --> model did not learn correctly --> model is under-segmenting

```json
{
    "reference": "ref.nii.gz",
    "prediction": "pred.nii.gz",
    "1.0": {
        "dsc": 0.0,
        "fbeta": 0,
        "nsd": 0.0,
        "vol_diff": 1.0,
        "rel_vol_diff": -100.0,
        "EmptyRef": false,
        "EmptyPred": true
    }
}
```

**Comment by naga-karthik** [comment]
Regarding 2:

> Empty reference and non-empty prediction 

The `100` is a bit weird given the the reference is empty so a division by an empty array should be `Infinity`, but I understand that we need to show that this is oversegmentation so you set 100 [here](https://github.com/valosekj/MetricsReloaded/pull/1/files#diff-971f87c8dd98f7eb7d9358d29874f84261c72d3ab739deb88d76fb41bc024707R905)

**Comment by valosekj** [comment]
Hey @naga-karthik! I believe the PR can now be beta-tested.

I implemented the following points based on our discussions:
- the script now accepts folders with multiple nii files as inputs (the compatibility with a single nii file is still preserved!)
- the output is saved as CSV (instead of JSON) --> easier to manipulate and plot
- the script also outputs a second CSV with mean +- std metrics
- output CSV files use full metric names (instead of metric abbreviations)

**Comment by naga-karthik** [comment]
So I am trying out the installation of MetricsReloaded and followed the commands mentioned [here](https://github.com/ivadomed/utilities/blob/b13a20a3939ae5b2fd133b62bbb1ad7a6aad273b/quick_start_guides/MetricsReloaded_quick_start_guide.md#installation). However, I get a `pip dependency` resolver error post-installation

<details>
<summary> Error </summary>

Screenshot 2024-03-20 at 9 58 34 PM

</details>

Is it something you also had? How did you solve this? @valosekj

**Comment by valosekj** [comment]
> So I am trying out the installation of MetricsReloaded and followed the commands mentioned [here](https://github.com/ivadomed/utilities/blob/b13a20a3939ae5b2fd133b62bbb1ad7a6aad273b/quick_start_guides/MetricsReloaded_quick_start_guide.md#installation). However, I get a `pip dependency` resolver error post-installation

I just tried the [commands](https://github.com/ivadomed/utilities/blob/b13a20a3939ae5b2fd133b62bbb1ad7a6aad273b/quick_start_guides/MetricsReloaded_quick_start_guide.md#installation) and the installation completed successfully on my end. We can check it together in person.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -34,7 +34,12 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install -r dataset_conversion/requirements.txt
+          git clone https://github.com/ivadomed/MetricsReloaded.git
+          cd MetricsReloaded
+          git checkout jv/add_rel_vol_error_metric
```
Comment (valosekj) [inline .github/workflows/ci.yml:39]: Remove this once https://github.com/ivadomed/MetricsReloaded/pull/2 is merged.

**Pull request merged**